Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -354,6 +516,35 @@ bool isFunctionJson(string fullName)
             }
         }
 
+        private async Task LocalAppFetch(IRepository repository, DeploymentInfoBase deploymentInfo, string targetBranch, ILogger logger, ITracer tracer)
+        {
+            var zipDeploymentInfo = (ZipDeploymentInfo)deploymentInfo;
+            var content = !string.IsNullOrEmpty(zipDeploymentInfo.ZipURL)
+                ? await DeploymentHelper.GetZipContentFromURL(zipDeploymentInfo, tracer)
+                : Request.Content;
+            //var content = Request.Content;
+            //var fileName = Path.ChangeExtension("app", deploymentInfo.FileExtention);
+            var tempDirPath = repository.RepositoryPath;
+            var targetInfo = FileSystemHelpers.DirectoryInfoFromDirectoryName(tempDirPath);
+            if (targetInfo.Exists)
+            {
+                var moveTarget = Path.Combine(targetInfo.Parent.FullName, Path.GetRandomFileName());
+                using (tracer.Step(string.Format("Renaming extractTargetDirectory({0}) to tempDirectory({1})", targetInfo.FullName, moveTarget)))
+                {
+                    targetInfo.MoveTo(moveTarget);
+                }
+            }          
+            var tempFilePath = Path.Combine(tempDirPath, deploymentInfo.FileName);
+            Directory.CreateDirectory(tempDirPath);
+            using (tracer.Step("Saving request content to {0}", tempFilePath))
+            {
+                await content.CopyToAsync(tempFilePath, tracer);
+                var cleanTask = Task.Run(() => DeleteFilesAndDirsExcept(tempFilePath, tempDirPath, tracer));
+                await Task.WhenAll(cleanTask);
+            }
+            CommitRepo(repository, zipDeploymentInfo);
```

Comment:
>CommitRepo [](start = 12, length = 10)

Is this needed?

Reply:
According to comments left in the code, "the deployment pipeline expects to see an least 1 commit in the IRepository." OneDeploy follows the existing deployment pipeline very closely after the files are copied to the temp repository so I assumed it would be needed.

Reply:
Add a brief comment explaining this in code.

// Need to commit because .. 
CommitRepo(repository, zipDeploymentInfo);

---
In reply to: [456093988](https://github.com/shrishrirang/kudu/pull/2#discussion_r456093988) [](ancestors = 456093988)

Reply:
Please add a comment in the code explaining why we need to commit.

---
In reply to: [456717046](https://github.com/shrishrirang/kudu/pull/2#discussion_r456717046) [](ancestors = 456717046,456093988)